src/chainstate/stacks/db/blocks.rs

```diff
@@ -898,12 +1025,23 @@ impl StacksChainState {
         Ok(Some(block))
     }
 
+    fn inner_load_block_header(block_path: &str) -> Result<Option<StacksBlockHeader>, Error> {
+        let sz = StacksChainState::get_file_size(block_path)?;
+        if sz == 0 {
+            debug!("Zero-sized block {}", &block_path);
+            return Ok(None);
+        }
+
+        let block_header: StacksBlockHeader = StacksChainState::consensus_load(block_path)?;
+        Ok(Some(block_header))
+    }
+
     /// Load up an anchored block header from the chunk store.
     /// Returns Ok(Some(blockheader)) if found.
     /// Returns Ok(None) if this block was found, but is known to be invalid
     /// Returns Err(...) on not found or I/O error
     pub fn load_block_header(
```

Comment:
Can `load_block_header` just invoke `inner_load_block_header` after computing the `block_path`?

Reply:
Done